src/zjs_fs.c

```diff
@@ -702,42 +648,19 @@ static jerry_value_t zjs_mkdir(const jerry_value_t function_obj,
                                const jerry_length_t argc,
                                uint8_t async)
 {
-    jerry_value_t js_cb = ZJS_UNDEFINED;
-    uint32_t mode = MODE_A_PLUS;
-    jerry_size_t size;
-    if (!jerry_value_is_string(argv[0])) {
-        return invalid_args();
+    // args: dirpath
+    ZJS_VALIDATE_ARGS(Z_STRING);
+
+#ifdef ZJS_FS_ASYNC_APIS
+    // async case adds callback arg
+    if (async) {
+        ZJS_VALIDATE_ARGS_OFFSET(1, Z_FUNCTION);
     }
-    if (argc > 2) {
-        if (async) {
-            if (!jerry_value_is_function(argv[2])) {
-                return invalid_args();
-            } else {
-                js_cb = argv[2];
-            }
-        }
-        if (!jerry_value_is_string(argv[1])) {
-            return invalid_args();
-        } else {
-            size = 4;
-            char mode_str[size];
+#endif
 
-            zjs_copy_jstring(argv[1], mode_str, &size);
-            if (!size) {
-                return zjs_error("size mismatch");
-            }
+    uint32_t mode = MODE_A_PLUS;
```

Comment:
Can remove mode, its unused. It was not actually being used for anything before but it was technically "used" so there was no compiler warning before.